### ISSUE TITLE
Query URL Param Session not refreshing when using Invoke-UDRedirect

### ISSUE THREAD

**Issue by jhadvan** [ironmansoftware/powershell-universal#3424]
### Version

4.3.0

### Severity

High

### Environment

IIS

### Steps to Reproduce

When I am using Invoke-UDRedirect -Url "/**Status**?**q=12345**" when going to Status page and using $Query.q to get the query parameters q it is returning null, but when I refresh the page the query parameter q is now declared.

When using UDRedirect /Status?q=12345 the value will be returned as NULL then refreshed q will now become 12345 then any future value will be returned as 12345

The query param works on some page, but on some page it doesn't

### Example Code

    New-UDTypography -Text 'Home' -Id 'homeText'
    New-UDButton -Text 'click me 123' -OnClick {
        Invoke-UDRedirect -Url "/parmPage?q=123"
    }

    New-UDButton -Text 'click me 321' -OnClick {
        Invoke-UDRedirect -Url "/parmPage?q=321"
    }

### Expected behavior

```console
I am expecting $Query.q to store the parameter of 12345
```

### Actual behavior

```console
$Query.Q is returning cache value of whatever is previously stored.
```

**Comment by jhadvan** [comment]
So I fixed this by removing the / when calling New-UDRedirect.

### Bad Code
New-UDTypography -Text 'Home' -Id 'homeText'
New-UDButton -Text 'click me 123' -OnClick {
    Invoke-UDRedirect -Url "/parmPage?q=123"
}

New-UDButton -Text 'click me 321' -OnClick {
    Invoke-UDRedirect -Url "/parmPage?q=321"
}

### Fixed Code

New-UDTypography -Text 'Home' -Id 'homeText'
New-UDButton -Text 'click me 123' -OnClick {
    Invoke-UDRedirect -Url "parmPage?q=123"
}

New-UDButton -Text 'click me 321' -OnClick {
    Invoke-UDRedirect -Url "parmPage?q=321"
}

**Comment by adamdriscoll** [comment]
I can't reproduce this with either code. Here's the full app code I'm trying. 

```
New-UDApp -Pages @(
    New-UDPage -Name Home -Content {
 New-UDTypography -Text 'Home' -Id 'homeText'
New-UDButton -Text 'click me 123' -OnClick {
    Invoke-UDRedirect -Url "/parmPage?q=123"
}


    }
    New-UDPage -Name parmPage -Content {
        New-UDButton -Text 'click me 321' -OnClick {
    Invoke-UDRedirect -Url "/parmPage?q=321"
}
        Show-UDToast $Query['q']
    }
)
```

**Comment by MikeShepard** [comment]
I have tested in msi (:5000) in 5.0 beta 7 and it works.
Nested IIS in 4.3 does not

**Comment by adamdriscoll** [comment]
Works for me in nested IIS. I am running the 4.3.1 nightly but no changes were made to this in this version. 

![image](https://github.com/ironmansoftware/issues/assets/1256531/c88095a5-ece4-40c1-a8a2-30546bdc2e5d)

![image](https://github.com/ironmansoftware/issues/assets/1256531/7694586e-0d55-4bf9-8c40-4bb8233e14f6)

**Comment by adamdriscoll** [comment]
Oh, I'm dumb. That's because this was fixed in 4.3.1 and not 4.3.0.
